cluster/juju/layers/kubernetes-worker/actions.yaml

```diff
@@ -14,4 +14,30 @@ microbot:
       delete:
         type: boolean
         default: False
-        description: Removes a microbots deployment, service, and ingress if True.
+        description: Remove a microbots deployment, service, and ingress if True.
+registry:
+     description: Create a private Docker registry
+     params:
+      htpasswd:
+        type: string
+        description: base64 encoded htpasswd file used for authentication.
+      htpasswd-plain:
+        type: string
+        description: base64 encoded plaintext version of the htpasswd file, needed by docker daemons to authenticate to the registry.
+      tlscert:
```

Comment:
As far as I know, you cannot get "valid" (signed by a trusted CA) certificates from the tls relation. And if you don't use a valid certificate, docker complains when pulling images, and rightly so. You can pass "--insecure-registry" to it via the docker_opts config option, but I'd like to avoid forcing users to do that.

I suppose I could fallback to asking the tls relation for a certificate if none is provided, and write something in the action output saying that "--insecure-registry" is required ? I'd like to consider this a "TODO" and deal with it in a separate PR, if that's OK for you.